com.unity.sg2/Editor/GraphDeltaRegistry/PreviewManager/HeadlessPreviewManager.cs

```diff
@@ -121,19 +121,40 @@ static Texture2D GenerateFourSquare(Color c1, Color c2)
             return tex;
         }
 
-        public HeadlessPreviewManager()
+        public void Initialize(string contextNodeName, Vector2 mainPreviewSize)
         {
             m_ErrorTexture = GenerateFourSquare(Color.magenta, Color.black);
             m_CompilingTexture = GenerateFourSquare(Color.blue, Color.blue);
             m_SceneResources = new PreviewSceneResources();
-        }
 
-        public void Initialize(string contextNodeName, Vector2 mainPreviewSize)
-        {
+            m_PreviewMaterialPropertyBlock = new();
             AddMainPreviewData(contextNodeName, mainPreviewSize);
             InitializeSRPIfNeeded();
         }
 
+        public void Cleanup()
+        {
+            if (m_ErrorTexture != null)
+            {
+                Object.DestroyImmediate(m_ErrorTexture);
+                m_ErrorTexture = null;
+            }
+
+            if (m_CompilingTexture != null)
+            {
+                m_CompilingTexture = GenerateFourSquare(Color.blue, Color.blue);
+                m_ErrorTexture = null;
+            }
+
+            if (m_SceneResources != null)
+            {
+                m_SceneResources.Dispose();
+                m_SceneResources = null;
+            }
+
+            m_PreviewMaterialPropertyBlock.Clear();
+        }
```

Comment:
Making a note to ask if per-preview cleanup also happens as resources are no longer.

Reply:
good point, there's a few more resources I need to cleanup here, most importantly the map of all the per-preview data 👍